addons/l10n_it_edi/models/account_edi_format.py

```diff
@@ -775,3 +774,315 @@ def _import_fattura_pa(self, tree, invoice):
             invoices += new_invoice
 
         return invoices
+
+    # -------------------------------------------------------------------------
+    # Export
+    # -------------------------------------------------------------------------
+
+    def _prepare_invoice_report(self, pdf_writer, edi_document):
+        self.ensure_one()
+        if self.code != 'fattura_pa':
+            return super()._prepare_invoice_report(pdf_writer, edi_document)
+        if edi_document.attachment_id:
+            pdf_writer.embed_odoo_attachment(edi_document.attachment_id)
+
+    def _is_compatible_with_journal(self, journal):
+        # OVERRIDE
+        self.ensure_one()
+        if self.code != 'fattura_pa':
+            return super()._is_compatible_with_journal(journal)
+        return journal.type in ('sale', 'purchase') and journal.country_code == 'IT'
+
+    def _is_required_for_invoice(self, invoice):
+        # OVERRIDE
+        self.ensure_one()
+        if self.code != 'fattura_pa':
+            return super()._is_required_for_invoice(invoice)
+
+        is_self_invoice = self._l10n_it_edi_is_self_invoice(invoice)
+        return (
+            (invoice.is_sale_document() or (is_self_invoice and invoice.is_purchase_document()))
+            and invoice.country_code == 'IT'
+        )
+
+    def _export_as_xml(self, invoice):
```

Comment:
We can keep it, but it is actually the same as _get_invoice_edi_content @dbkosky